### ISSUE TITLE
Suggested improvement to handling materials

### ISSUE THREAD

**Issue by BadDogSkyrim** [BadDogSkyrim/PyNifly#93]
Right now PyNifly loads the specular, splits the channel, plugs the red channel (Spec) into Blender's specular, and then Green (Gloss) into Blender's Metallic. The former is okay, but the latter is wrong. Green should be inverted and plugged into Roughness, since Roughness is literally just inverse of Glossiness.

In theory any FO4 materials using an envmap would also have the Spec (red) plugged into Blender's Metallic, but unless you can somehow read the .BGSM files to check for the "Environment Map" flag, I'd say it'd be safer to just leave Blender's Metallic node alone.
Also the normal map node only inverts the green channel by default, which makes sense because Blender wants OpenGL normals but FO4 uses DirectX ones. But since most FO4 normal maps are mainly saved as BC5 (i.e. they have no blue channel), you'd also need to invert the blue channel itself. I've had trouble with normals looking wonky because Blender expected a full white blue channel but got a full black one instead.
So basically:

For speculars, invert green channel and plug it into Roughness. (Also remove green channel from Metallic).

For normals, also invert the blue channel and not just the green one.

**Comment by thomasandersen** [comment]
A note on texture files is that Blender currently does not read all the DirectDraw Surface compression types. 
E.g. BC5 compression is not supported. IIRC BC5 is almost always used for the normal and specular/gloss/metallic maps in vanilla Fo4.

One solution from the top of my head would be to magically convert the files into something Blender can read.
E.g. using Texconv, https://github.com/Microsoft/DirectXTex/wiki/Texconv

Maybe a problem that should not be tackled by this add-on, but I understand it at the moment as full DDS support is not on Blender's priority at the moment

**Comment by BadDogSkyrim** [comment]
I don't have a good solution to the file format problem. Blender doesn't have full support even for the old dds formats--you can't paint onto one of those files, for example. Someone else has an issue open suggesting the use of Blender's default texture path setting. Maybe support a workflow where the user bulk converts from the game file tree to blender's texture path, and then if there's a png version of a dds referenced in the nif, load the png instead. But that seems cumbersome.

**Comment by thomasandersen** [comment]
Regarding reading bg\*m files. There is a CLI tool for converting a bg\*m to json
I think the Material Editor uses the tool/lib under the hood.

I currently understand it as the engine can actually read the json format too. I guess the binary bg\*m is for production.

Gibbed's Fallout 4 Tools at Nexus
https://www.nexusmods.com/fallout4/mods/1124/

SVN repo
http://svn.gib.me/public/fallout4/trunk

I think the author has a a git/hub repo too, but I am not sure if is maintained at the moment.

**Comment by BadDogSkyrim** [comment]
Closing this because the original suggestion is implemented. The other ideas are on the list for the future.
